lib/routes/v1.js

```diff
@@ -169,6 +183,18 @@ api.declare({
 
   let modDate = new Date();
 
+  // We want to make sure that all AMIs that we are submitting are valid
+  for (let region of input.regions) {
+    let validAmi = await amiExists(this.ec2, region.launchSpec.ImageId, region.region);
+    if (!validAmi) {
```

Comment:
API wrapper assumes handlers returns a promise... and if the promise is rejected, it'll return 500 with an incidentid, error message (which may be sensitive) is printed to log along with the incidentid.